linode/manager
# refactor: [M3-7376] - Region Select Autocomplete

## Description 📝
This PR's primary goal is to refactor the RegionSelect to use the new Autocomplete component.
It is a precursor to the work that needs to be done for DC Get Well. This refactor will help facilitating the addition of logic and API fetching within the RegionSelect.

I apologize for the diff size. It touches a lot of files that may not seem related, albeit benefited from the rework. It was hard to compartementalize some of the changes that felt natural to be a part of the PR. Please comment and ask for details as needed.

## Changes  🔄
- Replace the enhanced select with Autocomplete and configure to get parity
- Extract styles, types and utils for better separation of concerns
- Add relevant tests for extracted utils
- Replace all instanced of `selectedID` with `selectedId`
- Remove redundancy with Flag types typed as lowercase 
- Add a new e2e dedicated to the behavior of RegionSelect
- Update all relevant e2e tests
- Update all relevant unit tests

## Preview 📷
There should not be any significant visual change to the RegionSelect (maybe very minor things) and the behavior should be identical. 

**Note**: the only (intentional) change you should see is the ordering of the regions. The region ordering and DCs ordering is sorted alphabetically, except for "North America" which comes first (same as current RegionSelect)

## How to test 🧪

### Prerequisites
- Pull and run code locally

### Verification steps 
Verify the behavior and appearance of all all the places the RegionSelect is implemented (create flows, migration, clone, One click apps, Stackscripts etc)

## As an Author I have considered 🤔

*Check all that apply*
